src/main/java/com/redhat/rhjmc/containerjfr/platform/ServiceRef.java

```diff
@@ -41,36 +41,39 @@
  */
 package com.redhat.rhjmc.containerjfr.platform;
 
+import java.util.Optional;
+
+import javax.management.remote.JMXServiceURL;
+
 import org.apache.commons.lang.builder.EqualsBuilder;
 import org.apache.commons.lang.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
 public class ServiceRef {
 
-    private final String connectUrl;
-    private final String alias;
-    private final int port;
+    private final JMXServiceURL JMXServiceURL;
+    private Optional<String> alias = Optional.empty();
 
-    public ServiceRef(String connectUrl, int port) {
-        this(connectUrl, connectUrl, port);
+    public ServiceRef(JMXServiceURL JMXServiceURL) {
+        this.JMXServiceURL = JMXServiceURL;
     }
 
-    public ServiceRef(String connectUrl, String alias, int port) {
-        this.connectUrl = connectUrl;
-        this.alias = alias;
-        this.port = port;
+    public ServiceRef(JMXServiceURL JMXServiceURL, String alias) {
```

Comment:
Just a style thing but the constructor param should be named like `jmxServiceUrl` or `jmxServiceURL`, not `JMXServiceURL`.

Also if the other constructor has a `throws` declaration, they probably all should. In fact, this constructor should be the "primary" one, and the others should defer to it, ex:

```
public ServiceRef(JMXServiceURL jmxServiceUrl) {
  this(jmxServiceUrl, null);
}
```

That way as much logic is shared as possible.

Reply:
Makes sense!

Reply:
fixed.